src/sql/workbench/api/common/extHostDataProtocol.ts

```diff
@@ -916,8 +916,8 @@ export class ExtHostDataProtocol extends ExtHostDataProtocolShape {
 		return this._resolveProvider<azdata.designers.TableDesignerProvider>(handle).disposeTableDesigner(table);
 	}
 
-	public override $openTableDesigner(providerId: string, tableInfo: azdata.designers.TableInfo): Promise<void> {
-		this._proxy.$openTableDesigner(providerId, tableInfo);
+	public override $openTableDesigner(providerId: string, tableInfo: azdata.designers.TableInfo, serverInfo: azdata.ServerInfo): Promise<void> {
+		this._proxy.$openTableDesigner(providerId, tableInfo, serverInfo);
```

Comment:
Instead of passing over the entire server info object each time why not just pass the connection ID and do the lookup on the other side? There doesn't appear to be any need to pass this entire object over.

Reply:
I would even suggest adding an overload to `withServerInfo` that takes in the connection ID and does the lookup there. This would be useful to have in other contexts as well.